java/core/src/main/java/sleeper/core/record/process/status/ProcessRun.java

```diff
@@ -19,17 +19,17 @@
 import sleeper.core.record.process.RecordsProcessedSummary;
 
 import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 
 public class ProcessRun {
     private final String taskId;
-    private final ProcessRunStartedUpdate startedStatus;
-    private final ProcessFinishedStatus finishedStatus;
+    private final List<ProcessStatusUpdate> statusUpdates = new ArrayList<>();
 
     private ProcessRun(Builder builder) {
         taskId = Objects.requireNonNull(builder.taskId, "taskId must not be null");
-        startedStatus = Objects.requireNonNull(builder.startedStatus, "startedStatus must not be null");
-        finishedStatus = builder.finishedStatus;
+        statusUpdates.addAll(builder.statusUpdates);
```

Comment:
Using addAll makes it look like this might be mutable. It would be nice if we could be sure it's not. It would also be nice to have a descriptive error message if statusUpdates is null in the builder.